pkg/controller/volume/cache/desired_state_of_world.go

```diff
@@ -142,14 +146,15 @@ type volumeToAttach struct {
 	// volume and are scheduled to the underlying node. The key in the map is
 	// the name of the pod and the value is a pod object containing more
 	// information about the pod.
-	scheduledPods map[types.UniquePodName]pod
+	scheduledPods map[types.UniquePodName]Pod
 }
 
-// The pod object represents a pod that references the underlying volume and is
+// Pod represents a pod that references the underlying volume and is
 // scheduled to the underlying node.
-type pod struct {
-	// podName contains the name of this pod.
-	podName types.UniquePodName
+type Pod struct {
```

Comment:
Right, but you can extend the old `pod` object to store the new fields. Then you can have your new `GetPods` method return the new externally visible object that you construct using `pod` `volume` and `node` objects.